src/borg/_hashindex.c

```diff
@@ -39,6 +40,18 @@ typedef struct {
     int32_t num_buckets;
     int8_t  key_size;
     int8_t  value_size;
+}) HashHeader1;
+
+BORG_PACKED(
+typedef struct {
+    char magic[MAGIC_LEN];
+    int32_t version;
+    int32_t num_entries;
+    int32_t num_buckets;
+    int32_t num_empty;
+    int32_t key_size;
+    int32_t value_size;
```

Comment:
Is this a good opportunity to get rid of signed values for counts and sizes?

Reply:
could do that, but then the num_empty = -1 would need to get changed to e.g. 0xffffffff (which should be save, because we would make the HT smaller when having that many empty buckets).

Reply:
i checked the code and the internal in-memory data struct and the rest of the code also uses signed int.

thus, while it could be done, it is not in scope of this PR to refactor this.